### DISCUSSION TITLE
Removing quml library dependencies from "@project-sunbird/sunbird-questionset-editor"

### DISCUSSION THREAD

**Discussion by mithun30052001** [Sunbird-inQuiry/Community#106] [Improvements]
**Description:**
We are integrating the Quml Player as a web-components in the portal following the [documentation](https://github.com/project-sunbird/sunbird-quml-player?tab=readme-ov-file) in the [Issue](https://project-sunbird.atlassian.net/browse/ED-1473) and facing a dependency issue with `sunbird-questionset-editor`.

**Steps Taken:**
As a part of the implementation we removed the `"@project-sunbird/sunbird-quml-player"` and `"@project-sunbird/sunbird-quml-player-v9"` from `package.json.`

**Error Encountered:**
But since these Players have a dependency to the questionset-editor library we are getting the below error during the build

<img width="1332" alt="Screenshot 2024-01-30 at 2 18 30 PM" src="https://github.com/Sunbird-inQuiry/Community/assets/81811776/e291e9a3-3060-4614-ab4e-75451d35db5d">

**Request for Assistance**
So it would be really helpful if the dependency on quml library is removed for questionset-editor so that we could implement quml player as a web component

CC: @vinukumar-vs @rajeevsathish @princegupta1131

**Comment by rajnishdargan** [comment]
@mithun30052001,

The Questionset Editor currently relies on QuML Player for Preview functionality during the creation and curation process. It is essential to note that removing the QuML Player dependency from the Editor is not feasible at this time.

Before the release of version 7.0.0 of inQuiry, the QuML Player web component served as a dependency in the editor. 
Attempts were made to integrate both the `QuML Player web component` and `Questionset Editor web component` into the sunbirdEd-portal. However, during integration, an error occurred with the message: `Uncaught CKEditorError: ckeditor-duplicated-modules`. Unfortunately, a solution for this CKEditorError issue could not be identified at that moment. Refer to the CKEditor documentation for more details: [CKEditor Error Codes](https://ckeditor.com/docs/ckeditor5/latest/support/error-codes.html#error-ckeditor-duplicated-modules).

To move forward, resolving the CKEditorError issue in the sunbirdEd-portal is a prerequisite. Once a solution is identified, the QuML Player Library dependency can be replaced with the QuML Player web component in the Questionset Editor.

For testing and integrating the web components, it is recommended to use release-6.2.0 of inQuiry for testing purposes.

cc: @sajeshkayyath @PoojaJManjooran @vinukumar-vs @rajeevsathish @princegupta1131

**Reply by rajnishdargan** [reply]
@mithun30052001
Are you planning to replace only the quml-player library with the quml-player web component, or is there also a plan to replace the editor with web components?
If it's only the player, we can check the feasibility of replacing the quml-player library dependency with the quml-player web component in the editor.
cc: @sajeshkayyath @PoojaJManjooran @vinukumar-vs @rajeevsathish @princegupta1131

**Reply by mithun30052001** [reply]
Hi @rajnishdargan 

 We are indeed planning to replace only the quml-player library with the quml-player web component. However, we are facing a issue due to the existing dependency of the question set editor on the quml player. It would be nice if there is a feasible solution replacing the quml-player library dependency with the quml-player web component in the editor.
 
 cc: @sajeshkayyath @PoojaJManjooran @vinukumar-vs @rajeevsathish @princegupta1131

**Reply by rajnishdargan** [reply]
Hi @mithun30052001 
Thanks for the update, we will look into this.
cc: @sajeshkayyath @PoojaJManjooran @vinukumar-vs @rajeevsathish @princegupta1131

**Reply by PoojaJManjooran** [reply]
@mithun30052001 In which release are you planning to make this change? And is there a reason why you want to remove the QuML Player Library dependency from the question set editor?

**Reply by mithun30052001** [reply]
Hi @PoojaJManjooran
We are currently working on integrating the Quml web component into the SunbirdEd-portal version 7.0.0, planning to be implemented for the next release.So we are removing all the four player's(video,pdf,quml,epub) external library to make the portal light weight and removing the angular dependency while being used as direct plugin

**Comment by princegupta1131** [comment]
Hi @rajnishdargan as we mentioned above, we are facing an error while removing the QUML  package from the portal

It appears that the above packages are used as peer dependencies in the library, and their removal has caused these compilation errors.

Could you please provide the latest sunbird-questions-editor. version with this fix?

CC: @rajeevsathish @vinukumar-vs @mohitga  @sajeshkayyath @PoojaJManjooran

**Reply by rajnishdargan** [reply]
Hi @princegupta1131,

We have started the task for replacing the `"@project-sunbird/sunbird-quml-player"` dependency with `"@project-sunbird/sunbird-quml-player-web-component"`.
 Once this task is completed, we will update you.

**Reply by rajnishdargan** [reply]
Hi @princegupta1131,

We have replaced the dependency `"@project-sunbird/sunbird-quml-player"` with `"@project-sunbird/sunbird-quml-player-web-component"` in the question set editor and resource library. Below are the versions you can use for dev-testing:

`"@project-sunbird/sunbird-questionset-editor": "8.0.0-beta.3"`
`"@project-sunbird/sunbird-resource-library": "8.0.0-beta.2"`
`"@project-sunbird/sunbird-quml-player-web-component": "3.0.4"`

CC: @rajeevsathish @vinukumar-vs @mohitga @sajeshkayyath @PoojaJManjooran
